Xamarin.Forms.Core/VisualElement.cs

```diff
@@ -929,18 +936,24 @@ static void OnVisualChanged(BindableObject bindable, object oldValue, object new
 			(self as IPropertyPropagationController)?.PropagatePropertyChanged(VisualElement.VisualProperty.PropertyName);
 		}
 
-		static void FlowDirectionChanged(BindableObject bindable, object oldValue, object newValue)
+		static void FlowDirectionChanging(BindableObject bindable, object oldValue, object newValue)
```

Comment:
@samhouts I moved the EffectiveFlowDirection calculations here so that once the FlowDirection propertyChange fires the EffectiveFlowDirection has already been calculated